### DISCUSSION TITLE
Working with RoadRunner maps and scenarios in Carla

### DISCUSSION THREAD

**Discussion by haabe** [carla-simulator/carla#7185] [Q&A]
Hi,

I'm looking into a workflow of creating maps and scenarios in RoadRunner and replaying them in Carla.

The maps are exported as Datasmith exports, and imported using the RoadRunnerDatasmith import plugin, as recommended by Mathworks. I manually add the xodr file to the compilation.

The scenarios I export as a OpenScenario 1.x .xosx file, and then slightly modify to reflect the UE map instead of the OpenDRIVE map.

I have for a long time experienced that the cars won't spawn in their correct location in Carla, and that they won't drive as expected. Although for an earlier version of Carla, I saw [this description of the workflow where it is suggested to rotate maps 180 degrees](https://carla.readthedocs.io/en/0.9.4/how_to_make_a_new_map/) because of a bug in how Carla handles OpenDRIVE maps. With that in mind, I created a really simple map of a straight diagonal road in RR and exported it to Carla. I then created a simple scenario of a car driving along one of the lanes and exported it to OpenScenario. And, lo and behold, the car spawned in the  opposite end of my road. Rotating the map in Carla caused the car to almost spawn in the correct location. But I guess I didn't plan my simple road to be positioned perfectly centered at 0,0.

So my question, dear community, is how do we create a good workflow where we create maps in RR that can be easily rotated 180 degrees in Carla/UE and match the coordinates exactly so that they match up with the coordinates in the OpenScenario files?

**Comment by haabe** [comment]
To describe my issue a little better I've uploaded two screen captures of the same scenario running on the same map in RoadRunner and in Carla.

[![RoadRunner](https://img.youtube.com/vi/3ryeFrFwm3U/0.jpg)](https://www.youtube.com/watch?v=3ryeFrFwm3U)
[![Carla](https://img.youtube.com/vi/SX41VmRDSa4/0.jpg)](https://www.youtube.com/watch?v=SX41VmRDSa4)

As you can see, the ambulance is supposed to drive down the right lane. But in Carla, the ambulance spawns far into the scene, and takes a hard right and continues off the sidewalk.

**Reply by 3F999** [reply]
hi 
I have export a map .osm and converted by codes to .xodr but when I run it I recieve following error would you let me know how to solve this? Thank you very much.

line 249, in <module>
    lane_keeping_handler.connect_to_client()

line 70, in connect_to_client
    enable_mesh_visibility=True))
RuntimeError: opendrive could not be correctly parsed

**Reply by haabe** [reply]
@3F999 This seems totally unrelated to working with RR and Carla. But it seems like your OpenDRIVE file is corrupt. You should probably start looking into your OSM > OpenDRIVE conversion. Best of luck!

**Comment by 3F999** [comment]
Dear Havard,
thank you for your answer, actually the map is checked with opendrive viewer and has no problem and im working on windows is it possible to take a look on my program? with googlemeet or something?I will appreciate thatRegards,Behnaz    On Tuesday, February 27, 2024 at 08:30:14 AM GMT+1, Håvard Bartnes ***@***.***> wrote:  
 
 


@3F999 This seems totally unrelated to working with RR and Carla. But it seems like your OpenDRIVE file is corrupt. You should probably start looking into your OSM > OpenDRIVE conversion. Best of luck!
